packages/flame/test/components/position_component_test.dart

```diff
@@ -285,4 +295,301 @@ void main() {
+      expect(component.positionOf(Vector2(10, 0)), Vector2(55, 15));
+      expect(component.positionOf(Vector2(0, 10)), Vector2(45, 25));
+
+      expect(component.toLocal(Vector2(0, 0)), Vector2(-45, -15));
+      expect(component.toLocal(Vector2(50, 20)), Vector2(5, 5));
+      expect(component.toLocal(Vector2(55, 25)), Vector2(10, 10));
+      expect(component.toLocal(Vector2(100, 100)), Vector2(55, 85));
+    });
+
+    test('flips', () {
+      final component = PositionComponent()
+        ..size = Vector2(10, 10)
+        ..position = Vector2(50, 20)
+        ..anchor = const Anchor(0.6, 0.8);
+
+      expect(component.positionOf(Vector2(6, 8)), Vector2(50, 20));
+      expect(component.positionOf(Vector2(0, 0)), Vector2(44, 12));
+      component.flipHorizontally();
+      expect(component.positionOf(Vector2(6, 8)), Vector2(50, 20));
+      expect(component.positionOf(Vector2(0, 0)), Vector2(56, 12));
+      component.flipVertically();
+      expect(component.positionOf(Vector2(6, 8)), Vector2(50, 20));
+      expect(component.positionOf(Vector2(0, 0)), Vector2(56, 28));
+      component.flipHorizontally();
+      expect(component.positionOf(Vector2(6, 8)), Vector2(50, 20));
+      expect(component.positionOf(Vector2(0, 0)), Vector2(44, 28));
+      component.flipVertically();
+      expect(component.positionOf(Vector2(6, 8)), Vector2(50, 20));
+      expect(component.positionOf(Vector2(0, 0)), Vector2(44, 12));
+    });
+
+    test('center flips', () {
+      final component = PositionComponent()
+        ..size = Vector2(10, 10)
+        ..position = Vector2(50, 20)
+        ..anchor = const Anchor(0.6, 0.8);
+
+      expect(component.positionOf(Vector2(6, 8)), Vector2(50, 20));
+      expect(component.positionOf(Vector2(0, 0)), Vector2(44, 12));
+      expect(component.positionOf(Vector2(10, 0)), Vector2(54, 12));
+      component.flipHorizontallyAroundCenter();
+      expect(component.positionOf(Vector2(6, 8)), Vector2(48, 20));
+      expect(component.positionOf(Vector2(0, 0)), Vector2(54, 12));
+      expect(component.positionOf(Vector2(10, 0)), Vector2(44, 12));
+      component.flipVerticallyAroundCenter();
+      expect(component.positionOf(Vector2(6, 8)), Vector2(48, 14));
+      expect(component.positionOf(Vector2(0, 0)), Vector2(54, 22));
+      expect(component.positionOf(Vector2(10, 0)), Vector2(44, 22));
+    });
+
+    test('rotations', () {
+      final component = PositionComponent()
+        ..size = Vector2(8, 6)
+        ..position = Vector2(50, 20)
+        ..anchor = Anchor.center;
+
+      // Rotate the component in small increments counterclockwise
+      // and track the coordinate of its top-right corner
+      for (var i = 0; i < 30; i++) {
+        component.angle = -i / 10;
+        final cosA = math.cos(i / 10);
+        final sinA = math.sin(i / 10);
+        final expectedX = 50 + 5 * (0.8 * cosA - 0.6 * sinA);
+        final expectedY = 20 - 5 * (0.6 * cosA + 0.8 * sinA);
```

Comment:
I am a bit unclear where the 0.8/0.6 come from, is that the width/height divided by 10? and then times 5 to do half? if so can we simplify? if not can we have a comment explaining the reason

Reply:
sure, i've added some explanation here